src/Workspaces/Core/Portable/Shared/Extensions/ISymbolExtensions.cs

```diff
@@ -915,6 +915,8 @@ public static bool IsAwaitableNonDynamic(this ISymbol symbol, SemanticModel sema
             return getAwaiters.Any(VerifyGetAwaiter);
         }
 
+        public static bool IsValidGetAwaiter(this IMethodSymbol symbol) => VerifyGetAwaiter(symbol);
+
         private static bool VerifyGetAwaiter(IMethodSymbol getAwaiter)
         {
             var returnType = getAwaiter.ReturnType;
```

Comment:
seems super strange that *neither* of these methods is actually verifying that the name is actually 'GetAwaiter'.

Reply:
Really good point. I'll add that to the method I added since it's public